api/fasit.go

```diff
@@ -122,68 +214,232 @@ func fetchFasitResources(fasitUrl string, deploymentRequest NaisDeploymentReques
 	}
 	defer resp.Body.Close()
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		errorCounter.WithLabelValues("read_body").Inc()
-		return []byte{}, fmt.Errorf("Could not read body: %s", err)
+		return []byte{}, appError{err, "Could not read body", http.StatusInternalServerError}
+	}
+
+	httpReqsCounter.WithLabelValues(strconv.Itoa(resp.StatusCode), "GET").Inc()
+	if resp.StatusCode == 404 {
+		errorCounter.WithLabelValues("error_fasit").Inc()
+		return []byte{}, appError{err, "Resource not found in Fasit", http.StatusNotFound}
 	}
 
 	httpReqsCounter.WithLabelValues(strconv.Itoa(resp.StatusCode), "GET").Inc()
 	if resp.StatusCode > 299 {
 		errorCounter.WithLabelValues("error_fasit").Inc()
-		return []byte{}, fmt.Errorf("Fasit returned: %s (%s)", body, strconv.Itoa(resp.StatusCode))
+		return []byte{}, appError{err, "Error contacting Fasit)", resp.StatusCode}
 	}
 
 	return body, nil
 
 }
-func (fasit FasitClient) getScopedResource(resourcesRequest ResourceRequest, environment string, application string, zone string) (resource NaisResource, err error) {
-
+func (fasit FasitClient) getScopedResource(resourcesRequest ResourceRequest, fasitEnvironment, application, zone string) (NaisResource, AppError) {
 	req, err := fasit.buildRequest("GET", "/api/v2/scopedresource", map[string]string{
 		"alias":       resourcesRequest.Alias,
 		"type":        resourcesRequest.ResourceType,
-		"environment": environment,
+		"environment": fasitEnvironment,
 		"application": application,
 		"zone":        zone,
 	})
 
 	if err != nil {
-		return NaisResource{}, err
+		return NaisResource{}, appError{err, "unable to create request", 500}
 	}
 
-	body, err := fasit.doRequest(req)
-	if err != nil {
-		return NaisResource{}, err
+	body, appErr := fasit.doRequest(req)
+	if appErr != nil {
+		return NaisResource{}, appErr
 	}
 
 	var fasitResource FasitResource
 
 	err = json.Unmarshal(body, &fasitResource)
 	if err != nil {
 		errorCounter.WithLabelValues("unmarshal_body").Inc()
-		return NaisResource{}, fmt.Errorf("Could not unmarshal body: %s", err)
+		return NaisResource{}, appError{err, "Could not unmarshal body", 500}
 	}
 
-	resource, err = fasit.mapToNaisResource(fasitResource)
+	resource, err := fasit.mapToNaisResource(fasitResource)
 	if err != nil {
-		return NaisResource{}, err
+		return NaisResource{}, appError{err, "Unable to map response to Nais resource", 500}
 	}
-
 	return resource, nil
 }
 
+func (fasit FasitClient) createResource(resource ExposedResource, fasitEnvironment, application, zone, hostname string) (int, error) {
+	payload, err := json.Marshal(buildResourcePayload(resource, fasitEnvironment, zone, hostname))
+	if err != nil {
+		errorCounter.WithLabelValues("create_request").Inc()
+		return 0, fmt.Errorf("Unable to create payload (%s)", err)
+	}
+
+	req, err := http.NewRequest("POST", fasit.FasitUrl+"/api/v2/resources/", bytes.NewBuffer(payload))
```

Comment:
We have a helper function  doRequest(r *http.Request) ([]byte, AppError) that could be used to avoid http boilerplate code.